### PR TITLE
Feature/modal

### PR DESCRIPTION
**Pull request by ccollums**
**What (if any) features are you implementing?**
- added modal npm package 
- added react icons package 
- created modal pop up when a puzzle is clicked to view puzzle details and make a request 
**What (if anything) did you refactor?**
- N/A
**Were there any issues that arose?**
- N/A
**Is there anything needs to be prioritized?**
- N/A
**Next Steps?**
- Getting the API updated when a request is made, to update the users requests

### CODE REVIEW

File: src/components/Puzzle.tsx [hunk 1]
```diff
@@ -1,20 +1,47 @@
...
+  const closeModal = (event: React.MouseEvent) => {
+    event.preventDefault()
+    setModalOpen(false)
+  }
 
   return (
     <section>
-      <div className='individual-puzzle' onClick={() => console.log(id)}>
+      <div className='individual-puzzle' onClick={event => openModal(event)}>
         <img src={image} />
```
Comment (JRicardoT) [inline src/components/Puzzle.tsx:35]: We need to add an `alt` attribute to this image element.

File: src/components/Puzzle.tsx [hunk 2]
```diff
@@ -1,20 +1,47 @@
...
       </div>
       <h4 className='piece-count'>{pieceCount} pieces</h4>
+      <Modal
```
Comment (JRicardoT) [inline src/components/Puzzle.tsx:38]: We should add a property in the modal component to stop the scrolling in the background. We should also consider aria labels in the modal for screen readers.

File: src/components/PuzzleDetails.tsx [hunk 3]
```diff
@@ -1,12 +1,51 @@
...
+          <p>{price}</p>
+        </div>
+        <div className='paragraphs'>
+          <p className='bold'>Missing Pieces: </p> 
+          <p>{missingPieces}</p>
+        </div>
+       </div>
+      </div>
+      <div className='button-icon-flex'>
+        <IoClose size={70} onClick={event => closeModal(event)}/>
```
Comment (JRicardoT) [inline src/components/PuzzleDetails.tsx:46]: Let's change the cursor to a pointer when a user hovers over the closing component so it's more intuitive.

**Review by JRicardoT — APPROVED** [review]
This looks great. Thanks for the styling the Modal.

**Review by kbergsund — APPROVED** [review]
Excited about this! Thanks for the review, Ricardo.

**Pull request merged**